app/src/main/java/com/duckduckgo/app/browser/BrowserWebViewClient.kt

```diff
@@ -62,29 +58,54 @@ class BrowserWebViewClient(
     @Suppress("OverridingDeprecatedMember")
     override fun shouldOverrideUrlLoading(view: WebView, urlString: String): Boolean {
         val url = Uri.parse(urlString)
-        return shouldOverride(view, url, RequestOrigin.Unknown)
+        return shouldOverride(view, url)
     }
 
     /**
      * API-agnostic implementation of deciding whether to override url or not
      */
-    private fun shouldOverride(webView: WebView, url: Uri, requestOrigin: RequestOrigin): Boolean {
-        return try {
-            Timber.d("Request origin: ${requestOrigin.javaClass.simpleName}, shouldOverride $url")
+    private fun shouldOverride(webView: WebView, url: Uri): Boolean {
+        try {
+            Timber.v("shouldOverride $url")
 
             val urlType = specialUrlDetector.determineType(url)
-            val urlHandler = getUrlHandler(requestOrigin)
-            urlHandler.handleUrl(webView, urlType, webViewClientListener)
 
+            return when (urlType) {
+                is SpecialUrlDetector.UrlType.Email -> consume { webViewClientListener?.sendEmailRequested(urlType.emailAddress) }
```

Comment:
I could imagine the  consume function working but it doesn't add clarity here - it makes the code less readable as some sections return and others don't. I'd rather we follow a consistent pattern of returning directly or from somewhere else

Reply:
Done!